### PR TITLE
Fix search public address command test case error

### LINKED ISSUE #85: Test publicAddressSearch (Unit and integration)

### PR DESCRIPTION
**Pull request by travisim**
Fixes #85. This PR should have no test case errors and no Style errors

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/CommandParserTestUtil.java [hunk 1]
```diff
@@ -18,6 +18,9 @@ public static void assertParseSuccess(Parser<? extends Command> parser, String u
             Command expectedCommand) {
         try {
             Command command = parser.parse(userInput);
+            System.out.println("Actual: " + userInput);
+            System.out.println("Expected: " + expectedCommand);
```
Comment (Airiinnn) [inline src/test/java/seedu/address/logic/parser/CommandParserTestUtil.java:22]: I think you forgot to remove these print statements

File: src/test/java/seedu/address/logic/parser/SearchPublicAddressCommandParserTest.java
Comment (Airiinnn) [inline src/test/java/seedu/address/logic/parser/SearchPublicAddressCommandParserTest.java:1]: May want to include tests for duplicate prefixes.

Eg. `searchPublicAddress pa/smth1 pa/smth2`

Unless it's reusing a tested piece of coded that handles it?

File: src/test/java/seedu/address/logic/parser/SearchPublicAddressCommandParserTest.java [hunk 2]
```diff
@@ -3,28 +3,31 @@
...
     }
     @Test
     public void parse_missingCompulsoryField_failure() {
-        String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchPublicAddressCommand.MESSAGE_USAGE);
+        String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                SearchPublicAddressCommand.MESSAGE_USAGE);
         // no parameters
         assertParseFailure(parser, SearchPublicAddressCommand.COMMAND_WORD, expectedMessage);
-        // no public address
+        // no parameters + space
```
Comment (Airiinnn) [inline src/test/java/seedu/address/logic/parser/SearchPublicAddressCommandParserTest.java:30]: Instead of failing when there's a space, might be better to strip surrounding whitespaces like other commands

**Review by Airiinnn — APPROVED** [review]
Some minor changes that can be implemented in a future PR

**Review by TheRareFox — COMMENTED** [review]
Overall, LGTM. Good that there is comments and javadocs, and that styling and defensive coding is applied.

File: src/test/java/seedu/address/logic/commands/SearchPublicAddressCommandTest.java [hunk 3]
```diff
@@ -22,14 +21,14 @@ public class SearchPublicAddressCommandTest {
     @Test
     public void execute() {
```
Comment (Airiinnn) [inline src/test/java/seedu/address/logic/commands/SearchPublicAddressCommandTest.java:22]: Remember to add test for happy flow (and any other potential cases) as well. Possibly check for correctly handing for:
- Whitespaces
- Case sensitivity
- Partial public address
- Etc.

File: src/test/java/seedu/address/logic/parser/CommandParserTestUtil.java (hunk 1 above)
Comment (travisim) [reply src/test/java/seedu/address/logic/parser/CommandParserTestUtil.java:22]: thx sir

File: src/test/java/seedu/address/logic/parser/SearchPublicAddressCommandParserTest.java
Comment (travisim) [reply src/test/java/seedu/address/logic/parser/SearchPublicAddressCommandParserTest.java:1]: very good advice
